Classes/Neos/Neos/Ui/Domain/Model/Changes/AbstractMove.php

```diff
@@ -43,4 +44,17 @@ protected function finish(NodeInterface $node)
 
         parent::finish($this->getSubject());
     }
+
+    protected static function cloneNodeWithNodeData(NodeInterface $node)
+    {
+        if ($node instanceof Node) {
+            $originalNode = $node;
+            $node = clone $originalNode;
```

Comment:
Just wondering, shouldn' Node be aware how to clone itself properly? Maybe we should add it to `__clone` method of the Node class?

Reply:
@dimaip I am not sure; we could do it, but I would like to keep the side effects to a minimum.

All the best, Sebastian

Reply:
I'm not sure I understand what you mean here by side effects. I mean to do it in the Neos.ContentRepository package itself. I think it's only natural that when you clone an object it should be aware how to clone itself, without having to manually do it from the outside.
Anyways, do as you see fit here, I'm not experienced with PHP so much.